### PR TITLE
Add async methods to VectorStoreQATool

### PR DESCRIPTION
**Pull request by cbornet**

### CODE REVIEW

File: libs/community/langchain_community/tools/vectorstore/tool.py [hunk 1]
```diff
@@ -51,9 +54,28 @@ def _run(
         chain = RetrievalQA.from_chain_type(
             self.llm, retriever=self.vectorstore.as_retriever()
         )
-        return chain.run(
-            query, callbacks=run_manager.get_child() if run_manager else None
+        return chain.invoke(
```
Comment (cbornet) [inline libs/community/langchain_community/tools/vectorstore/tool.py:57]: change fron `run` to `invoke` as the former is deprecated.

**Review by baskaryan — APPROVED** [review]

File: libs/community/langchain_community/tools/vectorstore/tool.py [hunk 2]
```diff
@@ -51,9 +54,28 @@ def _run(
...
+        run_manager: Optional[AsyncCallbackManagerForToolRun] = None,
+    ) -> str:
+        """Use the tool asynchronously."""
+        from langchain.chains.retrieval_qa.base import RetrievalQA
+
+        chain = RetrievalQA.from_chain_type(
+            self.llm, retriever=self.vectorstore.as_retriever()
         )
+        return (
+            await chain.arun(
```
Comment (baskaryan) [inline libs/community/langchain_community/tools/vectorstore/tool.py:74]: ainvoke

Comment (cbornet) [reply libs/community/langchain_community/tools/vectorstore/tool.py]: done

File: libs/community/langchain_community/tools/vectorstore/tool.py (hunk 1 above)
Comment (alex-ber) [reply libs/community/langchain_community/tools/vectorstore/tool.py:57]: @cbornet 

This is incorrect fix. I get


`
AttributeError("'CallbackManager' object has no attribute 'ignore_chain'")
`
as root cause and
`
AttributeError: 'CallbackManager' object has no attribute 'raise_error'
`

The reason is that this code puts run_manager into **handlers**.

Please revert this change back. 

if you want to change from call chain.run() to chain.invoke() this is fine, but the code should be correect.

Comment (alex-ber) [reply libs/community/langchain_community/tools/vectorstore/tool.py:57]: @cbornet 
I've opened an issue https://github.com/langchain-ai/langchain/issues/18460

**Pull request merged**